bouncer/blacklist/views.py

```diff
@@ -27,3 +31,29 @@ def get(self, request):
             response = {"kind": "ip", "value": ip, "result": result}
             result_list.append(response)
         return Response(result_list)
+
+
+class PostRequestViewSet(viewsets.ViewSet):
+    @action(methods=["post"], detail=True)
+    def add_entry(self, request, pk=None):
+        data = json.loads(request.body)
+        for entry in data:
```

Comment:
This should work on parsed data, after validating the serializer. See response below.

You need to validate all the data first, otherwise if you have 3 objects and the 3rd one is invalid, you'll end up in:

1. Adding two objects
2. Stopping at the 3rd
3. Returning a 400 response, while you have partially created the given objects